OpenRA.Game/GameRules/ActorInfo.cs

```diff
@@ -125,33 +125,43 @@ public IEnumerable<ITraitInfo> TraitsInConstructOrder()
 			{
 				Trait = i,
 				Type = i.GetType(),
-				Dependencies = PrerequisitesOf(i).ToList()
+				Predecessors = PredecessorsOf(i).ToList(),
+				Requisites = RequisitesOf(i).ToList()
 			}).ToList();
 
-			var resolved = source.Where(s => !s.Dependencies.Any()).ToList();
+			var resolved = source.Where(s => !s.Predecessors.Any()).ToList();
 			var unresolved = source.Except(resolved);
 
 			var testResolve = new Func<Type, Type, bool>((a, b) => a == b || a.IsAssignableFrom(b));
-			var more = unresolved.Where(u => u.Dependencies.All(d => resolved.Exists(r => testResolve(d, r.Type))));
+			var more = unresolved.Where(u => u.Predecessors.All(d => !unresolved.Any(r => testResolve(d, r.Type))));
```

Comment:
This is the magic line in the resolving procedure.  While you're here, could you please add a comment explaining how it works?